lib/dialogs/toggle-settings-group.tsx

```diff
@@ -17,13 +17,19 @@ const ToggleGroup: FunctionComponent<Props> = ({
   isEnabled,
   onChange,
 }) => (
-  <ToggleControl
-    name={groupSlug}
-    value={slug}
-    id={`settings-field-${groupSlug}-${slug}`}
-    checked={isEnabled}
-    onChange={onChange}
-  />
+  <Fragment>
+    {isEnabled === null ? (
+      <span>Loading</span>
+    ) : (
+      <ToggleControl
+        name={groupSlug}
+        value={slug}
+        id={`settings-field-${groupSlug}-${slug}`}
+        checked={isEnabled}
+        onChange={onChange}
+      />
+    )}
```

Comment:
interesting approach; I didn't think of this.

if we haven't loaded the setting then I think we have to presume that sharing analytics is disabled and not send them. do you have an idea how to communicate that?

alternatively we might not have a good network connection but someone should theoretically be able to set this and let Simperium sync it. if they do that we should be able to use the value they set even though it hasn't synchronized with the server.

do you think it's worth the effort to let them toggle the setting in the loading state? flipped around, do you think it's worth preventing them from toggling the setting before it loads?